### PR TITLE
feat: Added azimuth and elevation calc

### PR DESCRIPTION
**Pull request by lsetiawan**
Added 'ecef2ae' function to compute azimuth and elevation. Additionally, 'azimuth' and 'elevation' properties are now added to the 'SolverTransponder' object on instantiation.

### CODE REVIEW

**Comment by lsetiawan** [comment]
Part of the work for https://github.com/uw-ssec/offshore-geodesy/issues/149

File: src/gnatss/utilities/geo.py [hunk 1]
```diff
@@ -47,3 +50,103 @@ def _get_rotation_matrix(
...
+    return az, elev
+
+
+def calc_enu_comp(
+    residuals: NDArray[Shape["*"], Float64],
+    az: NDArray[Shape["*"], Float64],
+    el: NDArray[Shape["*"], Float64],
+) -> NDArray[Shape["*"], Float64]:
+    """
+    Calculates the mean east, north, up components of the residuals
```
Comment (aniketfadia96) [inline src/gnatss/utilities/geo.py:127]:
```suggestion
    Calculates the mean east, north, up components of the residuals from N number of transponders.
```

File: src/gnatss/utilities/geo.py [hunk 2]
```diff
@@ -47,3 +50,103 @@ def _get_rotation_matrix(
...
+    from N number of transponders.
```
Comment (aniketfadia96) [inline src/gnatss/utilities/geo.py:128]:
```suggestion
```

**Review by aniketfadia96 — APPROVED** [review]
Added a minor formatting suggestion in the docstrings. LGTM overall.

File: src/gnatss/utilities/geo.py (hunk 2 above)
Comment (lsetiawan) [reply src/gnatss/utilities/geo.py:128]: Broke this up to the next line for character length spec

**Pull request merged**